packages/flutter/lib/src/material/scaffold.dart

```diff
@@ -2984,6 +2994,26 @@ class ScaffoldState extends State<Scaffold> with TickerProviderStateMixin, Resto
                   snackBarWidth: snackBarWidth,
                 ),
                 children: children,
+              ) : DecoratedBox(
+                decoration: widget.backgroundDecoration!,
+                child: CustomMultiChildLayout(
```

Comment:
I think here we can extract the `CustomMultiChildLayout` and its children out into a Widget and make it easier to read.

Reply:
Hey, when I extract out the `CustomMultiChildLayout` into a widget, some of the tests fail with this error message:

```
The following assertion was thrown while running async test code:
Golden "material.bottom_app_bar.custom_shape.2.png": Pixel test failed, 1.86% diff detected.
```

Any idea about this :thinking: